api/src/org/apache/cloudstack/api/command/admin/network/DedicateGuestVlanRangeCmd.java

```diff
@@ -106,14 +107,27 @@ public long getEntityOwnerId() {
 
     @Override
     public void execute() throws ResourceUnavailableException, ResourceAllocationException {
-        GuestVlan result = _networkService.dedicateGuestVlanRange(this);
-        if (result != null) {
-            GuestVlanRangeResponse response = _responseGenerator.createDedicatedGuestVlanRangeResponse(result);
-            response.setResponseName(getCommandName());
-            response.setObjectName("dedicatedguestvlanrange");
-            this.setResponseObject(response);
-        } else {
-            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, "Failed to dedicate guest vlan range");
+        if (projectId == null && (accountName == null || accountName.isEmpty()) ) {
```

Comment:
@kishankavala 
It is taken care inside the method call 
 Domain d = _domainDao.findById(domainId);
        if (d == null) {
            throw new InvalidParameterValueException("Please specify a valid domain");
        }